### PR TITLE
setting for auto save

### PR DESCRIPTION
**Pull request by theescapers**
Fixes # .

### Checklist

- [ ] I have included examples or tests
- [ ] I have updated the change log
- [ ] I am listed in the CONTRIBUTORS file
- [ ] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

-
-
-

### CODE REVIEW

File: src/CsvTextEditor/Constants.cs [hunk 1]
```diff
@@ -37,6 +37,11 @@ public static class AutomaticUpdates
     public static class Configuration
     {
         public const string CustomEditor = "Settings.Application.Editor.CustomEditor";
+        public const string TimedAutoSave = "Settings.Application.Editor.AutoSaveEditor";
```
Comment (GeertvanHorrik) [inline src/CsvTextEditor/Constants.cs]: Either the constant name or the literal is incorrect, shouldn't it be AutoSaveTime or something?

File: src/CsvTextEditor/Constants.cs [hunk 2]
```diff
@@ -37,6 +37,11 @@ public static class AutomaticUpdates
...
+
+        public static double AutoSaveTimeInSeconds { get { return 60.0; } }
+
+        public const bool AutoSaveDefault = false;
```
Comment (GeertvanHorrik) [inline src/CsvTextEditor/Constants.cs]: This should be the same as the setting, thus AutoSaveEditorDefaultValue and the other one should be called AutoSaveEditor

File: src/CsvTextEditor/ProjectManagement/Watchers/CsvTextEditorIsDirtyProjectWatcher.cs [hunk 3]
```diff
@@ -59,6 +76,23 @@ protected override Task OnActivatedAsync(IProject oldProject, IProject newProjec
             return base.OnActivatedAsync(oldProject, newProject);
         }
 
+        private void AutoSaveIfNeeded(object sender, EventArgs e)
+        {
+            if (!_configurationService.IsRoamingValueAvailable(Configuration.TimedAutoSave))
+                return;
```
Comment (GeertvanHorrik) [inline src/CsvTextEditor/ProjectManagement/Watchers/CsvTextEditorIsDirtyProjectWatcher.cs]: I don't think this needs to be here. Just use:

var autoSave = GetRoamingValue(configuration.AutoSaveEditor, configuration.AutoSaveEditorDefaultValue)

File: src/CsvTextEditor/ProjectManagement/Watchers/CsvTextEditorIsDirtyProjectWatcher.cs [hunk 4]
```diff
@@ -59,6 +76,23 @@ protected override Task OnActivatedAsync(IProject oldProject, IProject newProjec
...
+
+            if (_csvTextEditorInstance == null)
+                return;
```
Comment (GeertvanHorrik) [inline src/CsvTextEditor/ProjectManagement/Watchers/CsvTextEditorIsDirtyProjectWatcher.cs]: Use { and }

File: src/CsvTextEditor/ViewModels/StatusBarViewModel.cs [hunk 5]
```diff
@@ -139,9 +139,17 @@ private void OnCaretTextLocationChanged(object sender, CaretTextLocationChangedE
...
+            var indexOfNewLine = allText.IndexOf(Symbols.NewLineEnd);
+
+            if (indexOfNewLine != -1)
+            {
+                var firstLine = allText.Substring(0, indexOfNewLine);
+                var columnHeaders = firstLine.Split(Symbols.Comma);
+                return columnHeaders[location.Column.Index].Trim();
+            }
+
+            return "";
```
Comment (GeertvanHorrik) [inline src/CsvTextEditor/ViewModels/StatusBarViewModel.cs]: string.Empty

File: src/CsvTextEditor/Constants.cs (hunk 1 above)
Comment (Orcomp) [reply src/CsvTextEditor/Constants.cs]: AutoSaveInterval

File: src/CsvTextEditor/Constants.cs (hunk 2 above)
Comment (theescapers) [reply src/CsvTextEditor/Constants.cs]: Hi Geert, I've standardised everywhere on "AutoSaveEditor" but the GetRoamingValue still only ever seems to return false. I'm sure it's something really simply I've missed, can you see anything obvious that I've missed? 

Cheers

Garry

Comment (GeertvanHorrik) [reply src/CsvTextEditor/Constants.cs]: I just pushed a commit named "configuration changes". Just tested and that seems to work fine.

Comment (theescapers) [reply src/CsvTextEditor/Constants.cs]: OK, will check that out. Cheers.

**Pull request merged**